docs/cms/input-types.adoc

```diff
@@ -91,22 +91,42 @@ image::Content-selector-default.png[An input box with a dropdown options below i
   </config>
 </input>
 ----
-<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType Samples>>
+<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType>>
 <2> *allowPath* is used to limit the path of the content that may be selected using the <<allowPath Samples>>. By default, Content Selector
 only displays content from the same site.
 <3> *treeMode* toggles between tree and flat modes. Default value is `false` (flat mode). +
 image:Content-selector-tree.png[Input box with a dropdown with a tree structure, 350]
 <4> *hideToggleIcon* hides the toggle icon when set to `false`. +
 image:Content-selector-no-icon.png[Input box without an icon, 500]
 
-=== allowContentType Samples
+[#allowContentType]
+=== Allow content type pattern
+There are two modes of matching content type patterns in XP: `LEGACY` to simulate pattern matching before image:../images/xp-770.svg[XP 7.7.0,opts=inline] and `MATCH` as current default mode. It can be configured in <<../deployment/config.adoc#admin, config>>.
 
+==== Legacy +
+
+If pattern value can be found in any part of the content type name then it is accepted.
+Pattern will be extended by current application name if it's not set in <application-name:content-name> format or if none of ':', '*' symbols were used in pattern or pattern starts with '${app}'.
+
+==== Match +
+A pattern will be treated as Java Regular Expression and the whole content type name must match it to be accepted.
+It will be extended by the current application name if none of meta regular expression characters `\^$.|?*+()[{` , ':' were used.
+
+==== Special handling
+
+.'${app}' template expression
+Most content type names start with the name of the application where they are stored. It may be quite messy and hard to maintain the whole application name in every pattern, so the '${app}' template expression can be used to replace it.
+
+.'*' character
+There is a special handling of `*` symbol which differs from usual regex handling. It will be considered by parser as '0 or more of any character except linebreaks' meanwhile regex treat it as '1 or more of the preceding token'.
```

Comment:
> Also, why is the * character treated differently?

I guess we have to keep it as is, It was implemented long ago and many applications are written with pure `*` in expressions. Issue to follow-up
https://github.com/enonic/xp/issues/8578

Reply:
Cool, yeah. I thought about this a little bit last night, and it strikes me that there are a fair few issues you could run into that way. Would we consider changing it for XP 8? I don't know if it matters, but in the case that you do actually want the regular regex functionality of `*`, how would you do that? 

Anyway, back to the issue at hand: because `*` is treated differently, this should be clearly stated when the regexes are first brought up, I think.

Reply:
Out of topic here, I guess. Discuss it in the issue.

I added clarification earlier in the text. Is it OK?